runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -124,6 +124,12 @@ J9::Z::CodeGenerator::initialize()
       cg->setSupportsInlineEncodeASCII();
       }
 
+   static bool disableInlineVectorizedMismatch = feGetEnv("TR_disableInlineVectorizedMismatch") != NULL;
+   if (cg->getSupportsArrayCmpLen() && !disableInlineVectorizedMismatch)
```

Comment:
getSupportsArrayCmpLen would be disabled for arraylets and off-heap, right? Just confirming.

Reply:
https://github.com/eclipse/omr/pull/6983 enables `arraycmplen` in the same way `arraycmp` is enabled on each platform:
- AArch64 - disabled for arraylets, also disabled by environment variable `TR_aarch64DisableArrayCmpLen`
- Power - always enabled
- x86 - disabled for arraylets
- Z - always enabled

`arraycmplen` should probably be uniformly disabled for arraylets, and maybe by a single environment variable for all platforms. Same for `arraycmp`. I'll add a commit to https://github.com/eclipse/omr/pull/6983 to make `arraycmp` enablement more uniform and amend the commit introducing `arraycmplen`.

Not sure how off-heap will affect `arraycmp(len)` enablement, @VermaSh any input?